src/lidar_mini.hpp

```diff
@@ -18,28 +18,40 @@
  */
 class LIDARmini {
   private:
-    hwlib::pin_in &RX;
+    UARTConnection uart;
 
   public:
     /**
      * @brief Constructor
      */
-    LIDARmini(hwlib::pin_in &RX);
+    explicit LIDARmini();
 
-    std::array<char, 7> bytes;
+    /**
+     * @brief
+     * Returns an array with all the sensor data
+     *
+     * @description
+     * If the two startbytes are found returns the data package minus the 2 startbytes, 7 bytes as an array.
+     * If no startbytes are found with in a set amound of try's (100000) an empty array will be returned.
+     * The amound of try's has been determent by tests and 100000 works correctly. We suspect the reason why the amound is so high
+     * is because of the function needing to get in sick with the baudrate.
+     *
+     * @return
+     * std::array : bytes
+     */
+    std::array<char, 7> getSensorData();
 
     /**
      * @brief
-     * Returns the distance in cm
+     * Get distance in centimers
      *
      * @description
-     * Calculates the distance with the uart protocol of the data that comes in over the RX port in cm.
-     * Right now it returns the full package.
+     * Returns the distance to an object in centimeters
      *
      * @return
-     * char * : to the first element of the data package
+     * int : distance value
```

Comment:
No need to specify the return type, since doxygen takes it from the function, see [the wiki](https://github.com/R2D2-2018/R2D2-2018/wiki/Code-standard) for an example.